src/plugins/downloader/main/index.ts

```diff
@@ -155,20 +158,54 @@ export async function downloadSongFromId(
       (name: string) => (resolvedName = name),
       playlistFolder,
       trackId,
-      increasePlaylistProgress,
+      increasePlaylistProgress
     );
   } catch (error: unknown) {
     sendError(error as Error, resolvedName || id);
   }
 }
 
+function downloadSongOnFinishSetup({
+  ipc,
+  getConfig,
+}: Pick<BackendContext<DownloaderPluginConfig>, 'ipc' | 'getConfig'>) {
+  let currentUrl: string | undefined;
+  let duration: number | undefined;
+  let time = 0;
+
+  registerCallback(async (songInfo: SongInfo) => {
+    const config = await getConfig();
```

Comment:
Checking `config` in a callback is not a good idea (IO overhead)
How about calling `registerCallback` if `downloadOnFinish` is true?